scripts/upload_test_results/README.md

```diff
@@ -0,0 +1,49 @@
+# Auto-upload daily test results
+
+## Python script to automatically upload your HW Daily Test Results to the Github test_results repo 
+https://github.com/zephyrproject-rtos/test_results
+
+**To customize the script and run it on your machine, please modify code in Step 1 and Step 2 (Use Search to find that comments in the code of the script).**
+
+### Script description:
+1. The script creates a git directory, if a directory already created, script updates it (step 2-3) (-g argument)
+   * Fetches code from the source repo master branch zephyrproject-rtos/test_results
+   * Pushes updates to the origin repo master branch, in my case it is maksimmasalski/test_results
+2. Downloads versions.json file and finds the proper commit name for existing test results.
+   * When version found, it remembers its name and deletes versions.json file (-c argument)
+3. Creates a new branch, for example *intel_daily_master_177f2dac8b*
+4. Copies files from the Test results directory (-j argument) for a given commit to the .git repository *test_results*
+5. Renames all .xml files, for example from *junit-reel_board__zephyr_177f2dac8b.xml to reel_board.xml*
+6. The script does $git add for all test results files
+7. The script does $git commit
+8. The script does $git push 
+   * New branch will be created in your origin repo
+9. Checkouts back to the master branch. When you need to run the script again, all actions will be repeated again.
+Script execution finished.
+10. Now can open PR link from the terminal output or go to Github webpage and create the Pull Request to the upstream master branch. Github will generate a PR, you just click the "Compare & pull request" button on Github webpage.
+
+### Sanitycheck command sample:
+`sanitycheck --device-testing --hardware-map <map file> -p <platform> -T <tests path>`
+
+### Script usage:
+Please note that before you run script, you should enter your credentials to access Github in your Git.
+Export environment variable for GitPython log output:
+`export GIT_PYTHON_TRACE=full`
+Then run the script:
+`python3 upload_to_github.py -c <commit> -j <test results path> -g <git dir path/test_results> -t <commit title> -m <commit message>`
+
+### Example:
+`export GIT_PYTHON_TRACE=full`
+`python3 upload_to_github.py -c zephyr-v2.3.0-1307-gf014ba1ff1 -j /home/ztest2/work/testreport/sumreport/master/zephyr-v2.3.0-1307-gf014ba1ff1 -g /home/maksim/test_results -t "Master Daily Test Report" -m "Zephyr Daily Test Report for the master branch"`
+
+### Script has 3 required arguments:
+`-c` Upload daily test results of that commit<br/>
+`-j` Directory with all daily test result files<br/>
+`-g` Directory to place git repo test_results<br/>
+`-m` Commit message<br/>
+`-t` Commit title<br/>
```

Comment:
Are these optional or mandatory ?
They're inside section: "### Script has 3 required arguments:".
Either a new section for optional args should be added, either section title should be updated.

Reply:
@erwango they are mandatory.